cfgov/v1/models/home_page.py

```diff
@@ -381,6 +401,133 @@ class HomePageExcludedUpdates(models.Model):
     ]
 
 
+class HomePageCarouselItem(Orderable):
+    page = ParentalKey(
+        'v1.HomePage', on_delete=models.CASCADE, related_name='carousel_items'
+    )
+    title = models.CharField(max_length=45, help_text=(
+        "45 characters maximum (including spaces). "
+        "Sentence case, unless proper noun."
+    ))
+    body = models.TextField(max_length=160, help_text=(
+        "160 characters maximum (including spaces)."
+    ))
+    link_text = models.CharField(max_length=35, help_text=(
+        "35 characters maximum (including spaces). "
+        "Lead with a verb, and be specific."
+    ))
+    # TODO: Change this to use a URLField that also allows relative links.
+    # https://code.djangoproject.com/ticket/10896
```

Comment:
Either of those could work. There are some packages out there that do this ([for example](https://pypi.org/project/django-url-or-relative-url-field/)) as well.

I am not sure of the likelihood of using absolute URLs at all here. For example, we might want to link off-site ([order brochures](https://pueblo.gpo.gov/CFPBPubs/CFPBPubs.php), or to a YouTube video), in which case we need to support absolute URLs. On the other hand, if we allow that, editors will likely enter URLs that include https://www.consumerfinance.gov/, which we don't want (because it'd break links on other servers/locally).

The downside of the page chooser is that it's almost certain that we will want the link text to differ from the page title, which the page chooser doesn't easily support.

The use cases we probably need are:

1. Other Wagtail pages, in which case use a page chooser, but specify alternate text.
2. Other non-Wagtail pages on the site, in which case we want only relative links, never using e.g. https://www.consumerfinance.gov/. (To me the complexity of supporting both this and case 1 eliminate any UX improvement of the page chooser.)
3. Pages off-site, in which case we always want full URLs.

I'm not really sure what the best UX is here -- maybe a URL field that only supports absolute URLs if the domain isn't one of ours? Either way, using a plain CharField for now should be sufficient for the first implementation. Is there a more appropriate comment I can put for the TODO here to better sum this up?

Reply:
> The downside of the page chooser is that it's almost certain that we will want the link text to differ from the page title, which the page chooser doesn't easily support.

I still don't see this as a downside. You already have to have a separate field for the link text.

I suppose it'd be good to get a UX professional's take on it, but IMO, the best thing would be:

1. A page chooser for internal Wagtail pages
2. A `RegExField` for anything else, that validates that if the URL contains `consumerfinance.gov` it errors and requires you to make it relative
3. A link text field for either case

Reply:
Thanks for the writeup @Scotchester. I'm going to merge this as-is and open an internal issue to solicit further UX feedback on the chooser.